diff --git a/cs/checkpatch.pl b/cs/checkpatch.pl
index 4427572..705b0e1 100755
--- a/cs/checkpatch.pl
+++ b/cs/checkpatch.pl
@@ -4129,11 +4129,11 @@ sub process {
 			      $sl =~ /^\+\s+$Ident\s*:\s*\d+\s*[,;]/ ||
 			# other possible extensions of declaration lines
 			      $sl =~ /^\+\s+\(?\s*(?:$Compare|$Assignment|$Operators)/)) {
-				if (WARN("LINE_SPACING",
-					 "Missing a blank line after declarations\n" . $hereprev) &&
-				    $fix) {
-					fix_insert_line($fixlinenr, "\+");
-				}
+				# if (WARN("LINE_SPACING",
+				# 	 "Missing a blank line after declarations\n" . $hereprev) &&
+				#     $fix) {
+				# 	fix_insert_line($fixlinenr, "\+");
+				# }
 			}
 		}
 
